demo/src/main/java/io/weet/demo/models/UserModel.java

```diff
@@ -55,10 +55,10 @@ public void setPassword(String pwd) {
         this.password = pwd;
     }
 
-    // public void addAllergy(String name){
-    //     Allergen alg = new Allergen(name);
-    //     add = allergens.add(alg); //adds allergy to allergy list with name specified in method signature
-    // }
+    public void addAllergy(String name){
+        Allergen alg = new Allergen(name);
+        allergens.add(alg); //adds allergy to allergy list with name specified in method signature
+    }
 
     public void deleteAllergy(String name){

```

Comment:
This isn't needed. The user object should load the allergen data from DB so that it isn't manually added.

Reply:
I realized that these functions are used for `user.save`. Instead of adding/deleting individual allergen elements, I would use the `setAllergens` function to overwrite the whole list. This way duplicates are avoided.